src/main/java/ratpack/pac4j/internal/RatpackWebContext.java

```diff
@@ -20,40 +20,54 @@
 import com.google.common.collect.Maps;
 import com.google.common.collect.Sets;
 import io.netty.handler.codec.http.cookie.DefaultCookie;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
 import org.pac4j.core.context.Cookie;
 import org.pac4j.core.context.WebContext;
 import org.pac4j.core.context.session.SessionStore;
-import org.pac4j.core.exception.HttpAction;
+import org.pac4j.core.exception.TechnicalException;
+import org.pac4j.core.exception.http.HttpAction;
+import org.pac4j.core.profile.ProfileManager;
 import ratpack.exec.Promise;
 import ratpack.form.Form;
 import ratpack.form.internal.DefaultForm;
 import ratpack.form.internal.FormDecoder;
 import ratpack.handling.Context;
-import ratpack.http.*;
+import ratpack.http.HttpMethod;
+import ratpack.http.MediaType;
+import ratpack.http.Request;
+import ratpack.http.Response;
+import ratpack.http.TypedData;
 import ratpack.server.PublicAddress;
 import ratpack.session.Session;
 import ratpack.session.SessionData;
 import ratpack.util.MultiValueMap;
 
-import java.net.URI;
-import java.util.*;
-
 public class RatpackWebContext implements WebContext {
 
   private final Context context;
   private final RatpackSessionStore session;
+  private final ProfileManager profileManager;
   private final Request request;
   private final Response response;
   private final Form form;
+  private String body;
 
   private String responseContent = "";
 
   public RatpackWebContext(Context ctx, TypedData body, SessionData session) {
     this.context = ctx;
     this.session = new RatpackSessionStore(session);
+    this.profileManager = new ProfileManager(this, this.session);
     this.request = ctx.getRequest();
     this.response = ctx.getResponse();
-
+    this.body = body != null ? body.getText() : null;
```

Comment:
Is there a reason to store the text rather than the `TypedData`?

Reply:
My thinking is that storing the object might allow for better flexibility when body data is potentially needed in other ways in the future.

Reply:
Yeah, that's a good idea.  There's not any point at the moment but we might as well.